### DISCUSSION TITLE
Question on retrieving gas enthalpy (Python)

### DISCUSSION THREAD

**Discussion by ngom52** [equinor/neqsim#813] [Q&A] [answered]
Hi everyone, I recently started to use neqsim (Python version) to retrieve enthalpy of gas mixtures, but I noticed that the enthalpy values returned by NeqSim is significantky higher than the reference values that I have

I experimented on using neqsim to get the specific enthalpy of pure nitrogen at 400K using the following code:

from neqsim.thermo import fluid
gas = fluid('srk')
gas.setTemperature(400, "K")
gas.addComponent("nitrogen", 1)
gas.initThermoProperties()
print(gas.getEnthalpy("kJ/kmol"))

Using this code, the specific enthalpy returned is 7392.598679065902 kJ/kmol

When compared to an enthalpy table such as the one available at [https://www.ohio.edu/mechanical/thermo/property_tables/combustion/nitrogen_enth.html], the enthalpy at the same condition from the table is 2971 kJ/kmol

I hope that someone can point out the error in my code and suggest for improvements; thanks in advance!

**Comment by EvenSol** [comment, marked as answer]
Hi.
The absolute enthalpy will be dependent on reference temperature and if enthalpy of formation is included. In neqsim the reference temperature is 273.15 K (and enthalpy of formation is 0). It is therefore difficult to compare absolute enthalpies. See this [video with discussion](https://www.youtube.com/watch?v=pJMwOuWKUwg).

You should rather compare delta enthalpy between two states.

See the script below, giving a delta enthalpy of 1479.94 J/mol (for nitrogen at 10 bara when temperature is increased from 400K to 450K. If you compare delta enthalpies to other sources, they should be similar.

Supported units are 'J', 'J/mol', 'J/kg' and 'kJ/kg'
 
```
import neqsim
from neqsim.thermo import fluid, TPflash

gas = fluid('srk')
gas.setPressure(10.0, "bara")
gas.setTemperature(400, "K")
gas.addComponent("nitrogen", 1)
TPflash(gas)
gas.initProperties()
enthalpy1 = gas.getEnthalpy("J/mol")
print('enthalpy1 ', enthalpy1, 'J/mol')

gas.setTemperature(450.0, "K")
TPflash(gas)
gas.initProperties()
enthalpy2 = gas.getEnthalpy("J/mol")
print('enthalpy2 ', enthalpy2, 'J/mol')

deltaEnthalpy = enthalpy2 - enthalpy1

print('delta enthalpy ', deltaEnthalpy, 'J/mol')
```

Comparing to [ideal gas enthalpy tables](https://www.ohio.edu/mechanical/thermo/property_tables/combustion/nitrogen_enth.html) this seems fine. In the table 298K was used as reference temperature, but delta enthalpy will be similar.

**Reply by ngom52** [reply]
Hi, thanks a lot for the explanation, it was a great help!

**Comment by EvenSol** [comment]
In case you know the fluid will always be gas, you can skip the flash calculation, and you can speed up calculations as in the following script.

```
import neqsim
from neqsim.thermo import fluid, TPflash

gas = fluid('srk')
gas.addComponent("nitrogen", 1)
gas.setNumberOfPhases(1)
gas.setMaxNumberOfPhases(1)
gas.setForcePhaseTypes(True)
gas.setPhaseType(0, 'gas')

gas.setPressure(10.0, "bara")
gas.setTemperature(400, "K")
gas.initProperties()
enthalpy1 = gas.getEnthalpy("J/mol")
print('enthalpy1 ', enthalpy1, 'J/mol')

gas.setTemperature(450.0, "K")
gas.initProperties()
enthalpy2 = gas.getEnthalpy("J/mol")
print('enthalpy2 ', enthalpy2, 'J/mol')


deltaEnthalpy = enthalpy2 - enthalpy1


print('delta enthalpy ', deltaEnthalpy, 'J/mol')
```
